File: Javascript/0151-0200/167. Two Sum II - Input Array Is Sorted.java
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/?envType=study-plan-v2&envId=top-interview-150

var twoSum = function(numbers, target) {
    // Two-Pointers
    let first = 0, second = numbers.length - 1;
    while (first < second) {
        if(numbers[first] + numbers[second] == target) // target found
            return [first+1, second+1];
        else if(numbers[first] + numbers[second] < target)
            first++;
        else
            second--;
    }
    return null;
};

File: Python/0151-0200/167. Two Sum II - Input Array Is Sorted.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # Two-Pointers
        first, second = 0, len(numbers)- 1
        while first < second:
            if numbers[first] + numbers[second] == target: # target found
                return [first+1, second+1]
            elif numbers[first] + numbers[second] < target:
                first += 1
            else:
                second -= 1
                
        return None
  
